internal/test_cases/single_line_output_test_case.go

```diff
@@ -0,0 +1,101 @@
+package test_cases
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+// SingleLineOutputTestCase verifies a prompt exists, sends a command and matches the output against a string.
+type SingleLineOutputTestCase struct {
+	// The command to execute (the command's output will be matched against ExpectedPattern)
+	Command string
+
+	// ExpectedPattern is the regex that is evaluated against the command's output.
+	// Add \r\n at the end of the pattern if you're expecting a newline.
+	ExpectedPattern *regexp.Regexp
+
+	// ExpectedPatternExplanation is used in the error message if the ExpectedPattern doesn't match the command's output
+	ExpectedPatternExplanation string
+
+	// SuccessMessage is logged if the ExpectedPattern matches the command's output
+	SuccessMessage string
+}
+
+func (t SingleLineOutputTestCase) Run(shell *shell_executable.ShellExecutable, logger *logger.Logger) error {
+	promptTestCase := NewSilentPromptTestCase("$ ")
+
+	if err := promptTestCase.Run(shell, logger); err != nil {
+		return err
+	}
+
+	if err := shell.SendCommand(t.Command); err != nil {
+		return err
+	}
+
+	CRLFCondition := func(buf []byte) bool {
+		return len(buf) > 1 && bytes.Equal(buf[len(buf)-2:], []byte{'\r', '\n'})
+	}
+
+	output, err := shell.ReadBytesUntil(CRLFCondition)
+
+	// Whether the condition fails on not, we want to log the output
+	if len(output) > 0 {
+		shell.LogOutput(cleanOutput(output))
+	}
+
+	if err != nil {
+		// Here, we are sure we have read the entire output, so we don't read any more
+		if errors.Is(err, shell_executable.ErrConditionNotMet) {
+			return fmt.Errorf("Expected first line of output to end with '\\n' (newline), got %q", string(cleanOutput(output)))
+		} else if errors.Is(err, shell_executable.ErrProgramExited) {
+			return fmt.Errorf("Expected shell to be a long-running process, but it exited")
+		}
+		return err
+	}
+
+	// We will explicitly support the case where user uses CR/LF at the end
+	// and we receive CR/CR/LF
+	if len(output) > 2 && bytes.Equal(output[len(output)-3:], []byte{'\r', '\r', '\n'}) {
```

Comment:
Seems very specific to check for just these two cases, is there a more "general" way to handle this? 

For example, maybe something like `stripLineEnding` that takes bytes, and then:

- Collapses all consecutive `\r`s into a single `\r` (it doesn't matter how many there are really, they're equivalent)
- Removes `\r\n` from the end of the string

Reply:
(I say "remove \r\n" assuming that we won't have `ExpectedPattern` include it, separate comment above)